src/reader/scanner.cpp

```diff
@@ -33,27 +34,31 @@ std::unique_ptr<Token> Scanner::parse_symbol() {
 }
 
 std::unique_ptr<Numeral> Scanner::parse_numeral() {
-    int begin_index = current_index++;
+    int begin_index = ++current_index;
```

Comment:
![image](https://github.com/fedor-ivn/project-f/assets/10610844/843b11dd-c93f-4f0e-b4fe-bea16630fc57)

зря фёдор предложил тебе так переписать код, вот проблемы вылезли из-за инкремента и присвоения переменной одновременно

```suggestion
    int begin_index = current_index;
    ++current_index;
```

Reply:
а ещё лучше что-то такое сделать

```suggestion
    int begin_index = сurrent_index;
    
    if (source[current_index] == '+' || source[current_index] == '-') {
        ++current_index;
    }
```

и ниже

```cpp
bool has_integer_part = false;
while (std::isdigit(source[current_index])) {
    has_integer_part = true;
    ++current_index;
}
```